frontend/src/components/CustomerWeightsVisualization.tsx

```diff
@@ -0,0 +1,162 @@
+  const { t } = useTranslation();
+
+  if (customerStakes.length === 0)
+    return (
+      <div className={classes(css.stakesTooltip)}>
+        {t('No client weights set')}
+      </div>
+    );
+
+  return (
+    <div className={classes(css.stakesTooltip)}>
+      {!noTitle && <div>{t('Client weights')}</div>}
+      {showPercentageBar && (
+        <PercentageBar
+          stakes={customerStakes}
+          totalValue={totalValue}
+          width={15}
+        />
+      )}
+      {customerStakes.map((entry) => (
+        <div
+          key={entry.id}
+          className={classes(css.customer, {
+            [css.differsTooMuch]: entry.differsTooMuchFromPlanned,
+          })}
+        >
+          <Dot fill={entry.color} />
+          <span className={classes(css.name)}>{entry.name}</span>
+          <span className={classes(css.percentage)}>
+            {percent(1).format(totalValue > 0 ? entry.value / totalValue : 0)}
+          </span>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+const CustomerWeightsPercentageBar: FC<{
+  customerStakes: CustomerStakes[];
+  totalValue: number;
+  vertical?: boolean;
+  noTooltip?: true;
+  barWidth: number;
+}> = ({ customerStakes, totalValue, vertical, noTooltip, barWidth }) => {
+  if (noTooltip)
+    return (
+      <PercentageBar
+        stakes={customerStakes}
+        totalValue={totalValue}
+        vertical={vertical}
+        width={barWidth}
+      />
+    );
+  return (
+    <Tooltip
+      classes={{
+        arrow: classes(css.tooltipArrow),
+        tooltip: classes(css.tooltip),
+      }}
+      title={
+        <CustomerWeightsTooltipContent
+          customerStakes={customerStakes}
+          totalValue={totalValue}
+        />
+      }
+      placement="right"
+      arrow
+    >
+      <PercentageBar
+        stakes={customerStakes}
+        totalValue={totalValue}
+        vertical={vertical}
+        width={barWidth}
+      />
+    </Tooltip>
+  );
+};
+
+export const CustomerWeightsVisualization: FC<{
+  width: number;
+  height: number;
+  barWidth?: number;
+  noTooltip?: true;
+  vertical?: boolean;
+}> = ({ width, height, noTooltip, vertical, barWidth = 37 }) => {
+  const roadmapId = useSelector(chosenRoadmapIdSelector);
+  const { data: customers } = apiV2.useGetCustomersQuery(
+    roadmapId ?? skipToken,
+  );
+  const [data, setData] = useState<CustomerStakes[]>([]);
+  const totalValue = data.reduce((acc, { value }) => acc + value, 0);
+
+  useEffect(() => {
+    if (!customers) return;
+
+    setData(
+      [...customers]
+        .sort((a, b) => b.weight - a.weight)
+        .map(({ id, name, color, weight }) => ({
+          id,
+          name,
+          value: weight,
+          color,
+        })),
+    );
+  }, [customers, setData]);
+
+  return (
+    <div className={classes(css.stakes)}>
+      <div style={{ width, height }}>
+        <div className={classes(css.customerBarWithLink)}>
+          <CustomerWeightsPercentageBar
```

Comment:
Could CustomerStakesVisualization be used here instead of CustomerWeightsPercentageBar? It looks like the components are the same unless I missed something

Reply:
Currently its just slightly different tooltips